tests/test_section.py

```diff
@@ -61,6 +65,16 @@ def document_elm(self):
 
 class DescribeSection(object):
 
+    def it_provides_access_to_its_header(self, header_fixture):
+        section, Header_, header_ = header_fixture
+        header = section.header
+        Header_.assert_called_once_with('pretend element', section)
+
+        # access header again, dont instantiate with Header_ again, return same header
+        header = section.header
+        assert Header_.call_count == 1
+        assert header is header_
```

Comment:
Okay, here's what this one should look like. I'll provide commentary after the code:

```python
    def it_provides_access_to_its_header(self, header_fixture):
        section, Header_, sectPr, header_ = header_fixture
        header = section.header
        Header_.assert_called_once_with(sectPr)
        assert header is header_
```

As you can see, it's pretty close for the first one :)

The main thing driving this structure is what the `Header` object itself is going to look like. The `Header` object will be an element proxy, but that element is actually optional, as we know. All we can tell for sure is that a `w:sectPr` element will be available and if the `headerReference` element is present, it will be a child of that `w:sectPr`. So the "sure thing", i.e. the `w:sectPr`, is what we pass to the constructor. Everything else is figured out by the `Header` object when and if it needs it. So the constructor is quite simple. You get back a `Header` object that knows what it's `w:sectPr` is. Everything else happens later.